### PR TITLE
Make sure that the correct entrypoint file path is applied when bundling Pages applications

### LINKED ISSUE #4652: 🐛 BUG: Could not resolve "_worker.js" when running `wrangler pages dev` with `--proxy` flag
### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

3.22.1 [Wrangler]

### What version of Node are you using?

18.17.1

### What operating system are you using?

Mac

### Describe the Bug

I have a `_worker.js` in my project root:
```javascript=
export default {
  async fetch(request, env) {
    return new Response('Ok');
  },
}
```

I ran `pnpm exec wrangler pages dev --proxy <port>` and got:

```
✘ [ERROR] 1 error(s) and 0 warning(s) when compiling Worker.


▲ [WARNING] Failed to bundle _worker.js. Error: Build failed with 1 error:

  .wrangler/tmp/bundle-RCsC8p/middleware-insertion-facade.js:1:43: ERROR: Could not resolve
  "_worker.js"
      at failureErrorWithLog
  (/Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1636:15)
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1048:25
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1512:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    errors: [
      {
        detail: undefined,
        id: '',
        location: [Object],
        notes: [Array],
        pluginName: '',
        text: 'Could not resolve "_worker.js"'
      }
    ],
    warnings: []
  }
```

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

```
✘ [ERROR] 1 error(s) and 0 warning(s) when compiling Worker.


▲ [WARNING] Failed to bundle _worker.js. Error: Build failed with 1 error:

  .wrangler/tmp/bundle-RCsC8p/middleware-insertion-facade.js:1:43: ERROR: Could not resolve
  "_worker.js"
      at failureErrorWithLog
  (/Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1636:15)
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1048:25
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1512:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    errors: [
      {
        detail: undefined,
        id: '',
        location: [Object],
        notes: [Array],
        pluginName: '',
        text: 'Could not resolve "_worker.js"'
      }
    ],
    warnings: []
  }
```

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
Fixes #4652

### CODE REVIEW

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
I think that `entry.file` should already be resolved by the time we hit this code.
In most cases it is. From looking up the call stack I think the problem is in `packages/wrangler/src/pages/dev.ts`: if the working `directory` is not defined then the `script-path` (aka `singleWorkerScriptPath`) is used without resolving it correctly.

**Comment by dario-piotrowicz** [comment]
Thanks a bunch @petebacondarwin 🙂

I've applied the change in the `packages/wrangler/src/pages/dev.ts` hopefully that's what you had in mind please let me know how that looks 🙂

File: fixtures/pages-dev-proxy-with-script/tests/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				stdio: ["ignore", "ignore", "ignore", "ipc"],
+			}
+		).on("message", (message) => {
+			const parsedMessage = JSON.parse(message.toString());
+			resolve({
+				childProcess,
+				port: parsedMessage.port,
+			});
+		});
+		debugger;
```
Comment (petebacondarwin) [inline fixtures/pages-dev-proxy-with-script/tests/index.test.ts:49]: Remove?

**Review by petebacondarwin — APPROVED** [review]
I tested this locally by checking out the PR and running the following in the `pages-dev-proxy-with-script` fixture:

```
pnpm wrangler pages dev --proxy=9999 --script-path=custom/script/path/index.js
```

This fails with an error without the fix and works after the fix.

**Pull request merged**